server/server.go

```diff
@@ -73,6 +77,14 @@ type RPCServer struct {
 	// consumeSetting is configuration used when consuming from the message
 	// bus.
 	consumeSettings connection.ConsumeSettings
+
+	// consumerTags are used for graceful shutdown of the server. AMQP needs to
+	// know what consumers to stop sending messages to.
+	consumerTags []string
+
+	// These channels are used to signal shutdowns when calling Stop().
+	stopChan    chan struct{} // Closed when Close() is called.
+	stoppedChan chan struct{} // Close() will wait for this channel to close before exiting unless noWait is true.
```

Comment:
Is this channel used just to ensure that the **caller** of `Stop()` will wait til' the server is finished if it's the final call? If so I don't think it's needed, I think what's happening with the `WaitGroups` when `stopChan` gets a true value should always happen so any exit after `Stop()` is graceful.

I ask since if I understand correctly this channel is only used to hang in `Stop()` until the `WaitGroup` reaches 0, right?

Reply:
Yes it's used so that `Stop()` will wait until the server has finished the shutdown. It's always graceful as long as the main server will wait until `ListenAndServe` exists.

I actually think that it's more idiomatic to have `Stop` not wait at all.